crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -185,7 +179,45 @@ impl Solution {
     /// Check that the sum of tokens entering the settlement is not less than
     /// the sum of tokens exiting the settlement.
     fn verify_asset_flow(&self) -> Result<(), VerificationError> {
-        Ok(())
+        let mut flow: HashMap<eth::TokenAddress, num::BigInt> = Default::default();
+
+        // Interaction inputs represent flow into the contract, i.e. positive flow.
+        for input in self
```

Comment:
Isn't it actually the other way around? `input` in this context means tokens that get "used" for a given transaction meaning they leave the contract and the `output` flow back into the settlement contract.

Reply:
I also think that we defined `input` as inputs to the interaction (so not inputs into the settlement contract contract).